src/d2r_image/ocr.py

```diff
@@ -0,0 +1,255 @@
+        text = II_U.sub('U', ocr_output)
+    except:
+        # logging.error(f"Error _II_ -> _U_ on {ocr_output}")
+        text = ocr_output
+    # case: two 1's within a string; e.g., "S11PER MANA POTION"
+    try:
+        text = ONEONE_U.sub('U', text)
+    except:
+        # logging.error(f"Error _11_ -> _U_ on {ocr_output}")
+        pass
+    # case: an I within a number or by a sign; e.g., "+32I to mana attack rating"
+    try:
+        text = I_1.sub('1', text)
+    except:
+        # logging.error(f"Error I -> 1 on {ocr_output}")
+        pass
+    # case: a 1 within a string; e.g., "W1RT'S LEG"
+    try:
+        text = ONE_I.sub('I', text)
+    except:
+        # logging.error(f"Error 1 -> I on {ocr_output}")
+        pass
+    # case: a solitary I; e.g., " I TO 5 DEFENSE"
+    cnt = 0
+    while True:
+        cnt += 1
+        if cnt > 30:
+            # logging.error(f"Error ' I ' -> ' 1 ' on {ocr_output}")
+            break
+        if " I " in text:
+            text = text.replace(" I ", " 1 ")
+            continue
+        elif ' I\n' in text:
+            text = text.replace(' I\n', ' 1\n')
+            continue
+        elif '\nI ' in text:
+            text = text.replace('\nI ', '\n1 ')
+            continue
+        break
+    # case: a solitary S; e.g., " 1 TO S DEFENSE"
+    cnt = 0
+    while True:
+        cnt += 1
+        if cnt > 30:
+            # logging.error(f"Error ' S ' -> ' 5 ' on {ocr_output}")
+            break
+        if " S " in text:
+            text = text.replace(" S ", " 5 ")
+            continue
+        elif ' I\n' in text:
+            text = text.replace(' S\n', ' 5\n')
+            continue
+        elif '\nI ' in text:
+            text = text.replace('\nS ', '\n5 ')
+            continue
+        break
+    # case: consecutive I's; e.g., "DEFENSE: II"
+    repeat = False
+    cnt = 0
+    while "II" in text:
+        cnt += 1
+        if cnt > 30:
+            # logging.error(f"Error 4 on {ocr_output}")
+            break
+        text = text.replace("II", "11")
+        repeat = True
+        repeat_count += 1
+    if repeat and repeat_count < 10:
+        _fix_regexps(text)
+    return text
+
+
+def _check_known_errors(text):
+    for key in ERROR_RESOLUTION_MAP:
+        if key in text:
+            text = text.replace(key, ERROR_RESOLUTION_MAP[key])
+    return text
+
+def _ocr_result_dictionary_check(
+    original_text: str,
+    confidences: list,
+    word_list: list = all_words(),
+    normalized_lev_threshold: float = 0.6,
+    ocr_confidence_threshold: float = 0.9
+    ) -> str:
+    confidences = [x/100 for x in confidences]
+    if all(x >= ocr_confidence_threshold for x in confidences):
+        return original_text
+    #print(f"{original_text} {confidences}")
+    text = original_text.replace('\n', ' NEWLINEHERE ')
+    corrected_text = ""
+    word_count = 0
+    for word in text.split(' '):
+        word = word.strip()
+        if word and word != "NEWLINEHERE":
```

Comment:
Another minor stylistic preference, feel free to dismiss this from me: I kind of prefer to put the shorter end of an if statement at the top in a method like this.  When I see this, I'm immediately thinking "ok, so we need to handle word is None, word is False, and word == "NEWLINEHERE" as special cases possibly", but then I see at the bottom that we're basically skipping this if word is None? I would prefer to explicity handle the single cases here like:

```python
if word is None:
  next # This seems to be the current behavior
elif word == "NEWLINEHERE"
  corrected_text += "\n"
else:
  <main method>
```

In this case, it's actually seems likely to me that there's no way `word` could actually be `None` or `False`, or even `''`, so I think it would be safe to drop the top part, but the general goal of these [guard clause](https://deviq.com/design-patterns/guard-clause) style if statements is to avoid nesting if clauses unecessarily by putting the short ones that can `return`, `break` or `next` at the top, and trying to keep most of it going through the general case.  But I can also see why you might have wanted to put the meat of the function at the top and handle the exception at the bottom.

Reply:
Yes, I like this suggestion. I think you're right that it won't happen, but I still kept `if not word: continue`. I can't remember if when I made this function if I ran into some weird case and had to put it there. But yes otherwise making changes as suggested.